### PR TITLE
Mvcintegration

### PR DESCRIPTION
**Pull request by VolhaMan**
@manusovich 
Spring MVC has been implemented

### CODE REVIEW

File: src/main/java/weather/OpenWeatherWeatherGetter.java [hunk 1]
```diff
@@ -15,14 +19,20 @@
     private String applicationId;
     private String url;
 
+    @Autowired
+    @Value("4671654")
```
Comment (manusovich) [inline src/main/java/weather/OpenWeatherWeatherGetter.java]: Use properties file

File: src/main/java/weather/WeatherController.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * Created by volha on 8/27/16.
+ */
+
+@RestController
+public class WeatherController {
+
+    private WeatherGetterManager manager;
+    private DataBaseProcedures dataBaseProcedures;
```
Comment (manusovich) [inline src/main/java/weather/WeatherController.java]: There is one word Database. You should not make B as capital.

File: src/main/java/weather/WeatherController.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+
+    @Autowired
+    public WeatherController(WeatherGetterManager manager, DataBaseProcedures dataBaseProcedures) throws WeatherException {
+        this.manager = manager;
+        this.dataBaseProcedures = dataBaseProcedures;
+    }
+
+    @RequestMapping(value = "/weather", method = RequestMethod.GET, produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE})
+    public @ResponseBody CurrentWeather handle() throws WeatherException {
```
Comment (manusovich) [inline src/main/java/weather/WeatherController.java]: Don't put annotations in one line with method signature

File: src/main/java/weather/WeatherController.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    private DataBaseProcedures dataBaseProcedures;
+
+
+    @Autowired
+    public WeatherController(WeatherGetterManager manager, DataBaseProcedures dataBaseProcedures) throws WeatherException {
+        this.manager = manager;
+        this.dataBaseProcedures = dataBaseProcedures;
+    }
+
+    @RequestMapping(value = "/weather", method = RequestMethod.GET, produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE})
```
Comment (manusovich) [inline src/main/java/weather/WeatherController.java]: I think it is not necessary to specify produces here. No?

File: src/main/webapp/WEB-INF/web.xml [hunk 5]
```diff
@@ -3,4 +3,17 @@
...
+        <servlet-name>dispatcher</servlet-name>
+        <servlet-class>org.springframework.web.servlet.DispatcherServlet</servlet-class>
+        <load-on-startup>1</load-on-startup>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>dispatcher</servlet-name>
+        <url-pattern>/</url-pattern>
+    </servlet-mapping>
+    <servlet-mapping>
+        <servlet-name>default</servlet-name>
```
Comment (manusovich) [inline src/main/webapp/WEB-INF/web.xml]: What is that?

**Comment by VolhaMan** [comment]
@manusovich all fixes according to comments have been done. Additionally history has been moved to MVC as well.

File: src/main/java/weather/WeatherController.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    @Autowired
+    public WeatherController(WeatherGetterManager manager, DataBaseProcedures dataBaseProcedures) throws WeatherException {
+        this.manager = manager;
+        this.databaseProcedures = dataBaseProcedures;
+    }
+
+    @RequestMapping(value = "/weather", method = RequestMethod.GET)
+    public
+    @ResponseBody
+    CurrentWeather handleWeather() throws WeatherException {
```
Comment (manusovich) [inline src/main/java/weather/WeatherController.java]: When i said that it should not be in one line, i mean annotation. So it has to be something like 

```
@ResponseBody
@RequestMapping(value = "/weather", method = RequestMethod.GET)
public CurrentWeather handleWeather() throws WeatherException {
```

File: src/main/webapp/WEB-INF/web.xml (hunk 5 above)
Comment (manusovich) [inline src/main/webapp/WEB-INF/web.xml]: ?

File: src/main/java/weather/WeatherController.java (hunk 6 above)
Comment (VolhaMan) [inline src/main/java/weather/WeatherController.java]: This annotation is part of method definition. It marks that returning object should be include in response

Comment (manusovich) [inline src/main/java/weather/WeatherController.java]: I am not asking what is that. All i am saying that you should not put annotation between "public" and "CurrentWeather"

File: src/main/webapp/WEB-INF/web.xml (hunk 5 above)
Comment (VolhaMan) [inline src/main/webapp/WEB-INF/web.xml]: It is default servlet. It serves static resources. Without it javascript mis not accessible.

Comment (manusovich) [inline src/main/webapp/WEB-INF/web.xml]: Please use mvc resources to define resource folders and mapping to have better integration with spring. http://www.baeldung.com/spring-mvc-static-resources

File: src/main/java/weather/WeatherController.java (hunk 6 above)
Comment (VolhaMan) [inline src/main/java/weather/WeatherController.java]: Fixed

**Comment by VolhaMan** [comment]
All issues havebeen fixed

**Pull request merged**